src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,65 @@
-export const Pagination = () => {};
+import { getNumbers } from '../../utils';
+
+type Props = {
+  total: number;
+  perPage: number;
+  currentPage: number;
+  onPageChange: (page: number) => void;
+};
+
+export const Pagination: React.FC<Props> = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  const pagesNumber = Math.ceil(total / perPage);
+  const paginated = getNumbers(1, pagesNumber);
+
+  return (
+    <ul className="pagination">
+      <li className={`page-item ${currentPage === 1 && ('disabled')}`}>
+        <a
+          data-cy="prevLink"
+          className="page-link"
+          href="#prev"
+          aria-disabled={currentPage === 1}
+          onClick={() => {
+            if (currentPage !== 1) {
+              onPageChange(currentPage - 1);
+            }
+          }}
+        >
+          «
+        </a>
+      </li>
+      {paginated.map(page => (
+        <li key={page} className={`page-item ${currentPage === page && ('active')}`}>
+          <a
+            data-cy="pageLink"
+            className="page-link"
+            href={`#${page}`}
+            onClick={() => onPageChange(page)}
+          >
+            {page}
+          </a>
+        </li>
+      ))}
+      <li className={`page-item ${currentPage === pagesNumber && ('disabled')}`}>
+        <a
+          data-cy="nextLink"
+          className="page-link"
+          href="#next"
+          aria-disabled={currentPage === pagesNumber}
+          onClick={() => {
+            if (currentPage !== pagesNumber) {
+              onPageChange(currentPage + 1);
```

Comment:
Pls move the function inside `onClick` handler to a separate const and use as a callback here